### ISSUE TITLE
Keep API key secret

### ISSUE THREAD

**Issue by gracekfan** [tamuhack-org/moo#63]

**Comment by SaltyQuetzals** [comment]
This will need to be accomplished by using a [templating engine](https://expressjs.com/en/resources/template-engines.html) (I'd recommend using Handlebars, personally). 

What will need to happen is:

1. We'll need to use [this function](https://github.com/tamuhack-org/moo/blob/86879094f5932fddb9bffc42c2131bd8970e8500/src/getEvents.ts) I built at the start of Moo.
2. We'll need to modify the workshops page to use templates rather than raw HTML. (
3. The output of the `getEvents` function will need to be provided as context to the template. Something like...
```
// In workshopsRoute
const events = await getEvents();
res.render('workshops.hbs', {events});
```
4. In the template, use the [`{{#each}}`](https://handlebarsjs.com/block_helpers.html) directive (CTRL+F for "Simple Iterators") to map the events list to DOM.

**Comment by eightants** [comment]
I'm still not too sure how to implement this. Which package should I use and how should I implement the templates? 
What does getEvents return and how should I get it into variables?
The way I implemented the events page, I get the list of events then make another call to get detailed info for each event, then, I make a third call to get the name of the venue (because eventbrite uses venue_ids instead). How would this work with the getEvents function?

**Comment by SaltyQuetzals** [comment]
Very similarly!

The backend of Moo uses TypeScript (which is similar to JavaScript, just
with enforced types for scalability).

I provided the `getEvents` function as a convenience, but using it is not
necessary, especially if you've performed more complex tasks with the event
information.

My recommendation is to push up your branch (including the API key, we'll
destroy it and replace the way you get it with something more secure before
merging). After that, I can help you convert your JS code into TypeScript
code, which we'll use on the backend.

Once we get your event-retrieval code converted, we'll want to render the
workshop template on the client side. This is accomplished by providing the
events you've retrieved as variables to the templating engine. This sounds
complex when I write it out here, but in code, it's rather simple.

[This example](https://expressjs.com/en/guide/using-template-engines.html)
uses a different templating language than what we'll use, but focus more on
the `app.get` part.

**Comment by SaltyQuetzals** [comment]
Very similarly!

The backend of Moo uses TypeScript (which is similar to JavaScript, just
with enforced types for scalability).

I provided the `getEvents` function as a convenience, but using it is not
necessary, especially if you've performed more complex tasks with the event
information.

My recommendation is to push up your branch (including the API key, we'll
destroy it and replace the way you get it with something more secure before
merging). After that, I can help you convert your JS code into TypeScript
code, which we'll use on the backend.

Once we get your event-retrieval code converted, we'll want to render the
workshop template on the client side. This is accomplished by providing the
events you've retrieved as variables to the templating engine. This sounds
complex when I write it out here, but in code, it's rather simple.

[This example](https://expressjs.com/en/guide/using-template-engines.html)
uses a different templating language than what we'll use, but focus more on
the `app.get` part.

**Comment by eightants** [comment]
Implemented with #65
